controller/rest/system.go

```diff
@@ -2017,6 +2017,45 @@ func handlerSystemGetRBAC(w http.ResponseWriter, r *http.Request, ps httprouter.
 		NvCrdSchemaErrors:        acceptable[4],
 	}
 
+	fedRole := cacher.GetFedMembershipRoleNoAuth()
+	if (fedRole == api.FedRoleMaster && (acc.IsFedReader() || acc.IsFedAdmin() || acc.HasPermFed())) ||
+		(fedRole == api.FedRoleJoint && acc.HasGlobalPermissions(share.PERMS_CLUSTER_READ, 0)) {
+		otherAlerts := map[string]string{}
+		// _fedClusterLeft(206), _fedClusterDisconnected(204)
+		//disconnectedStates := utils.NewSet(_fedClusterLeft, _fedClusterDisconnected)
+		var ids map[string]bool
+		if fedRole == api.FedRoleMaster {
+			ids = cacher.GetFedJoinedClusterIdMap(acc)
+		} else {
+			if m := cacher.GetFedMasterCluster(acc); m.ID != "" {
+				ids = map[string]bool{
+					m.ID: true,
+				}
+			}
+		}
+		if len(ids) > 0 {
+			for id, _ := range ids {
+				s := cacher.GetFedJoinedClusterStatus(id, acc)
+				if elapsed := time.Since(s.LastConnectedTime); s.LastConnectedTime.IsZero() || elapsed > (time.Duration(_teleFreq)*time.Minute) {
+					var alert string
+					if fedRole == api.FedRoleMaster {
+						clusterInfo := cacher.GetFedJoinedCluster(id, acc)
+						alert = fmt.Sprintf("Managed cluster %s is disconnected from primary cluster", clusterInfo.Name)
```

Comment:
minor: The same string shows up in both rest/system.go and rest/auth.go.  Given that these string have to be the same for the acceptedAlerts to work, maybe make the string as a constant?

Reply:
use getFedDisconnectAlert() to generate the alert/key